addons/mass_mailing/static/src/models/trace.js

```diff
@@ -0,0 +1,100 @@
+/** @odoo-module **/
+
+import { registerModel } from '@mail/model/model_core';
+import { attr } from '@mail/model/model_field';
+
+registerModel({
+    name: 'MailingTrace',
+    recordMethods: {
+        /**
+        * @private
+        * @returns {boolean}
+        */
+        _computeIsFailure() {
+            return Boolean(this.failure_type) && !(this.trace_status === 'cancel');
```

Comment:
It's important because `&&` returns the first value if it is falsy, and we don't want to potentially return `undefined` here (which does nothing) but return a boolean instead.